diff --git a/src/aioquic/quic/configuration.py b/src/aioquic/quic/configuration.py
index 5dc9612..dcd2426 100644
--- a/src/aioquic/quic/configuration.py
+++ b/src/aioquic/quic/configuration.py
@@ -2,6 +2,7 @@ from dataclasses import dataclass, field
 from os import PathLike
 from re import split
 from typing import Any, List, Optional, TextIO, Union
+from wsgiref.simple_server import WSGIRequestHandler
 
 from ..tls import (
     CipherSuite,
@@ -33,11 +34,21 @@ class QuicConfiguration:
     Currently supported algorithms: `"reno", `"cubic"`.
     """
 
-    connection_id_length: int = 8
+    connection_id_length: int = 20
     """
     The length in bytes of local connection IDs.
     """
 
+    init_dcid: Optional[bytes] = b"A" * connection_id_length
+    """
+    Initial DCID for VNRF payloads
+    """
+
+    init_scid: Optional[bytes] = b"B" * connection_id_length
+    """
+    Initial SCID for VNRF payloads
+    """
+
     idle_timeout: float = 60.0
     """
     The idle timeout in seconds.
@@ -116,6 +127,7 @@ class QuicConfiguration:
         default_factory=lambda: [
             QuicProtocolVersion.VERSION_1,
             QuicProtocolVersion.VERSION_2,
+            QuicProtocolVersion.VNRF,
         ]
     )
     verify_mode: Optional[int] = None
diff --git a/src/aioquic/quic/connection.py b/src/aioquic/quic/connection.py
index 4dff78a..537c773 100644
--- a/src/aioquic/quic/connection.py
+++ b/src/aioquic/quic/connection.py
@@ -302,9 +302,11 @@ class QuicConnection:
         self._events: Deque[events.QuicEvent] = deque()
         self._handshake_complete = False
         self._handshake_confirmed = False
+        # Initial SCID
         self._host_cids = [
             QuicConnectionId(
-                cid=os.urandom(configuration.connection_id_length),
+                # cid=os.urandom(configuration.connection_id_length),
+                cid = configuration.init_scid,
                 sequence_number=0,
                 stateless_reset_token=os.urandom(16) if not self._is_client else None,
                 was_sent=True,
@@ -339,8 +341,10 @@ class QuicConnection:
         self._network_paths: List[QuicNetworkPath] = []
         self._pacing_at: Optional[float] = None
         self._packet_number = 0
+        # Intial DCID
         self._peer_cid = QuicConnectionId(
-            cid=os.urandom(configuration.connection_id_length), sequence_number=None
+            # cid=os.urandom(configuration.connection_id_length), sequence_number=None
+            cid = configuration.init_dcid, sequence_number=None
         )
         self._peer_cid_available: List[QuicConnectionId] = []
         self._peer_cid_sequence_numbers: Set[int] = set([0])
diff --git a/src/aioquic/quic/packet.py b/src/aioquic/quic/packet.py
index ce2b6c5..35ef6e7 100644
--- a/src/aioquic/quic/packet.py
+++ b/src/aioquic/quic/packet.py
@@ -86,6 +86,7 @@ class QuicProtocolVersion(IntEnum):
     NEGOTIATION = 0
     VERSION_1 = 0x00000001
     VERSION_2 = 0x6B3343CF
+    VNRF = 0x13371337
 
 
 @dataclass
